### ISSUE TITLE
Error after HACS install  ha-rpi_rf

### ISSUE THREAD

**Issue by hawkeyeP** [markvader/ha-rpi_rf#9]
I have core-2022.3.8 installed with supervisor-2022.03.5 and Home Assistant OS 7.6.
I installed via HACS, rebooted.  I am getting the following errors:
Any ideas what might be causing this failure?
Thank you!

```
ERROR (SyncWorker_1) [homeassistant.util.package] Unable to install package RPi.GPIO==0.7.1: error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      running bdist_wheel
      running build
      running build_py
      running build_ext
      building 'RPi._GPIO' extension
      error: command 'gcc' failed: No such file or directory
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for RPi.GPIO
  error: subprocess-exited-with-error
  
  × Running setup.py install for RPi.GPIO did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      running install
      /usr/local/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      running build_ext
      building 'RPi._GPIO' extension
      error: command 'gcc' failed: No such file or directory
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  WARNING: No metadata found in /usr/local/lib/python3.9/site-packages
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> RPi.GPIO

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
WARNING: You are using pip version 22.0.3; however, version 22.0.4 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
```

**Comment by markvader** [comment]
I'll have a look into it and come back to you.  Looks like an issue with the RPi.GPIO dependency (although nothing has changed with it recently)

Did you have the official RPI RF integration installed before it was deprecated?

**Comment by brnomendes** [comment]
The same error happened to me. Home Assistant 2022.3.8
I already installed the deprecated version once, but I no longer have it installed.

**Comment by markvader** [comment]
I need to rewrite the integration to containerize it, as when the next update comes out I think we will lose out on access to the gpio layer.

**Comment by markvader** [comment]
I haven't been able to reproduce this error on my system but can see that there are some problems with the rpi_gpio library failing on systems that are missing python-dev and/or gcc
Can you both let me know what hardware, you are running on and what installation method you have use - Home Assistant Operating System, Docker Container, Home Assistant Core.

Thanks
Mark

**Comment by hawkeyeP** [comment]
I have the all in one solution which I believe is home assistant os.  It is running on a raspberry pi 4.

**Comment by markvader** [comment]
Ah, same as me.  Would you mind copying your system info from Configuration / Settings / Info (Theres a copy icon where you can copy in github format).  You can delete anything additional like integrations installed/Hacs/spotify if they have their own sections. Basicly its the System health and Home Assistant Supervisor sections i'm interested in to see if theres a common version that is missing the gcc compiler

## System Health

version | core-2022.3.8
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.9
os_name | Linux
os_version | 5.10.103-v8
arch | aarch64

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 7.6
-- | --
update_channel | stable
supervisor_version | supervisor-2022.03.5
docker_version | 20.10.9
supported | failed to load: Unsupported (mine is unsupported because i run additional containers not installed through home assistant add-on store)
board | rpi4-64
supervisor_api | ok
version_api | ok

</details>

**Comment by hawkeyeP** [comment]
## System Health

version | core-2022.3.8
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.9
os_name | Linux
os_version | 5.10.103-v7l
arch | armv7l
timezone | America/Denver


<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 7.6
-- | --
update_channel | stable
supervisor_version | supervisor-2022.03.5
docker_version | 20.10.9
disk_total | 57.8 GB
disk_used | 6.1 GB
healthy | true
supported | true
board | rpi4
supervisor_api | ok
version_api | ok
installed_addons | Samba share (9.5.1), AppDaemon 4 (0.5.1), Mosquitto broker (6.0.1), SSH & Web Terminal (8.1.0), Check Home Assistant configuration (3.10.0), chrony (2.0.1)

</details>

**Comment by markvader** [comment]
It's looking like a Home Assistant OS 32bit Vs 64bit issue. Although I still need to work out what it needs to fix it. 

Would you mind trying to install the rpi_gpio integration in HACS and confirming if there is an issue there as well. Both use the same library but I updated this one to latest release however the rpi_gpio integration is on the previous release.

**Comment by hawkeyeP** [comment]
My installation was via hacs.

**Comment by brnomendes** [comment]
[markvader](https://github.com/markvader) just to let you know, my settings (it's aarch64):

## System Health

version | core-2022.3.8
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.9
os_name | Linux
os_version | 5.10.103-v8
arch | aarch64

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 7.6
-- | --
update_channel | stable
supervisor_version | supervisor-2022.03.5
docker_version | 20.10.9
disk_total | 58.7 GB
disk_used | 5.7 GB
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | Studio Code Server (4.2.0), Terminal & SSH (9.3.0)

</details>

**Comment by brnomendes** [comment]
> It's looking like a Home Assistant OS 32bit Vs 64bit issue. Although I still need to work out what it needs to fix it.
> 
> Would you mind trying to install the rpi_gpio integration in HACS and confirming if there is an issue there as well. Both use the same library but I updated this one to latest release however the rpi_gpio integration is on the previous release.

Yes, I tried installing rpi_gpio integration and it worked here.
https://github.com/thecode/ha-rpi_gpio

**Comment by brnomendes** [comment]
@markvader, the rpi_gpio integration in HACS uses RPi.GPIO version `0.7.1a4` instead of `0.7.1`
I tried make a copy of rpi_rf with RPi.GPIO==0.7.1a4 and it worked

**Comment by markvader** [comment]
> @markvader, the rpi_gpio integration in HACS uses RPi.GPIO version `0.7.1a4` instead of `0.7.1`
> I tried make a copy of rpi_rf with RPi.GPIO==0.7.1a4 and it worked


Thanks for testing that. I have done another beta release with this change, available in HACS if you enable the beta release checkbox

v2022.4.0-beta2

**Comment by hawkeyeP** [comment]
The beta2 allowed the switches to show up in HA, but it doesn't seem to be sending out the codes to turn the switches on and off.  I've rebooted several times.

**Comment by markvader** [comment]
> The beta2 allowed the switches to show up in HA, but it doesn't seem to be sending out the codes to turn the switches on and off. I've rebooted several times.

Thanks for the report, I'll get testing/debugging it tonight

**Comment by markvader** [comment]
I haven't been able to get back to this in a few days, but hope to be able to get back to it soon, in the mean time if anyone is in a position to debug the issue feel free to contribute.

**Comment by inferusvv** [comment]
Is it connected to error `Platform error switch.rpi_rf - Requirements for rpi_rf not found: ['RPi.GPIO==0.7.1'].` when I am trying to reload Home Assistant?

**Comment by arniebarni** [comment]
> Is it connected to error `Platform error switch.rpi_rf - Requirements for rpi_rf not found: ['RPi.GPIO==0.7.1'].` when I am trying to reload Home Assistant?

This seems to be related to the beginning of this thread. It could be circumvented by requiring RPi.GPIO==0.7.1a4 as in the latest beta release:

> @markvader, the rpi_gpio integration in HACS uses RPi.GPIO version `0.7.1a4` instead of `0.7.1` I tried make a copy of rpi_rf with RPi.GPIO==0.7.1a4 and it worked

While this allows for clicking of the buttons/switches, they do not work on the hardware side. So there must be another issue at hand.

**Comment by arniebarni** [comment]
After some debugging and testing older versions - also of this repository - I found that the introduction of the `length` parameters caused the troubles.  The order of `length` and `pulselength` was inconsistent. Feel free to merge my PR #20 or merge the fix yourself :-)

**Comment by markvader** [comment]
Thank you for debugging and spotting this @arniebarni (That'll teach me not to add code when using a small phone screen).
I have merged this and released https://github.com/markvader/ha-rpi_rf/releases/tag/2022.4.0b3 If others could test it would be much appreciated.  Then after its confirmed working i'll release as 2022.4.1 and then look at bringing RPi.GPIO version 0.7.1 back in again.

**Comment by markvader** [comment]
Should say, it's working for me.

**Comment by hawkeyeP** [comment]
Working for me!  Thank you.

**Comment by yellowpanther** [comment]
> Thank you for debugging and spotting this @arniebarni (That'll teach me not to add code when using a small phone screen). I have merged this and released https://github.com/markvader/ha-rpi_rf/releases/tag/2022.4.0b3 If others could test it would be much appreciated. Then after its confirmed working i'll release as 2022.4.1 and then look at bringing RPi.GPIO version 0.7.1 back in again.

Also worked for me. Thanks a lot

**Comment by markvader** [comment]
The credit goes to @arniebarni, but also thanks to all who submitted comments.
I have released [v2022.4.1](https://github.com/markvader/ha-rpi_rf/releases/tag/2022.4.1) 

...and [v2022.4.2b1](https://github.com/markvader/ha-rpi_rf/releases/tag/2022.4.2b1)
2022.4.2b1 contains the updated 0.7.1 RPI.gpio library which currently does not install correctly (so avoid it)
I'll close this issue for now and continue trying to see if I can get 0.7.1 to work as a requirement
